src/ecalc/libraries/libecalc/common/libecalc/core/models/compressor/train/fluid.py

```diff
@@ -233,11 +232,9 @@ def mix_in_stream(
             eos_model=self.fluid_model.eos_model,
         )
 
-        mixed_fluid = FluidStream(
-            existing_fluid=_neqsim_fluid_stream,
-            fluid_model=self.fluid_model,
+        return FluidStream(
+            existing_fluid=mixed_neqsim_fluid_stream,
+            temperature_kelvin=temperature_kelvin,
+            pressure_bara=pressure_bara,
+            fluid_model=dto.FluidModel(composition=mixed_fluid_composition, eos_model=self.fluid_model.eos_model),
```

Comment:
should EOS model be a part of the fluid stream object? should we consider moving it out and only specify when we do something with the streams? in theory self and other can have different EOS models here...

Reply:
Good question..

@Jfoul and I discussed the possibility of deviating FluidModels, I think we concluded that it won't cause an issue? You only care about the composition of the two fluids. We only set a new fluid_model for the mixed fluid stream. As it calculates its properties at initialization

Reply:
In theory we should not mix two fluids with different EOS's, but it wouldn't cause a direct error since you use the EOS from the original stream (?). Maybe we should include a check though to make sure that the users doesn't try mix two different EOSs (might result in a breaking change for some users though?).

Reply:
Should leave for the "consistency and check/validation PR" as well then.

Reply:
I guess we would not want to e.g. set a EOS for ALL fluid models for a given eCalc model. We might ahve some with GERG and some with SRK within the same eCalc model..right?